src/helpers/internal_viewport.ts

```diff
@@ -153,7 +151,62 @@ export class InternalViewport {
     }
   }
 
-  adjustPositionX(targetCol: HeaderIndex) {
+  setViewportOffset(offsetX: Pixel, offsetY: Pixel) {
+    this.setViewportOffsetX(offsetX);
+    this.setViewportOffsetY(offsetY);
+  }
+
+  adjustViewportZone() {
+    this.adjustViewportZoneX();
+    this.adjustViewportZoneY();
+  }
+
+  /**
+   *
+   * @param zone
+   * @returns Computes the absolute coordinate of a given zone inside the viewport
+   */
+  getRect(zone: Zone): Rect | undefined {
+    const targetZone = intersection(zone, this);
```

Comment:
open question: is it worth to lose the git blame history just so the private and public methods are grouped together ?

Reply:
> open question: is it worth to lose the git blame history just so the private and public methods are grouped together ?

Let's close the question: no, it's not worth it. It's much better to keep the diff history

That's one of the reason I don't like "sections" of code  with titles like `// private` where some methods are grouped together. The sections inevitably ends up being wrong.

Moving the methods for the sake of regrouping them also makes the review harder. Especially since it's mixed with other cleaning/fixes. I have to carefully compare each line before/after to check it anything changed